javascript/ql/lib/semmle/javascript/Modules.qll

```diff
@@ -111,6 +111,23 @@ abstract class Module extends TopLevel {
   cached
   abstract DataFlow::Node getAnExportedValue(string name);
 
+  /**
+   * Gets a value that is exported as the whole exports object of this module.
+   */
+  cached
+  DataFlow::Node getABulkExportedNode() { none() } // overridden in subclasses
```

Comment:
Done.

Note that there was some dodgy behaviour in `getAnExportFromModule` regarding the use of type inference:
```js
  result.analyze().getAValue() = mod.(AmdModule).getDefine().getAModuleExportsValue()
```
AFAICT, if the exported value was potentially `undefined`, then _any_ node that could potentially be `undefined` would be seen as being an export from that AMD module 😖. (Likewise for any other abstract value, such as `null`)

It's hard to predict what effect this will have, and we might lose some results that were previously caught by chance because of this. For now I'll just evaluate on default slugs again, but maybe we should run something larger.

Reply:
> AFAICT, if the exported value was potentially undefined, then any node that could potentially be undefined would be seen as being an export from that AMD module confounded.

Oops. Luckily that can only happen if there exists such an `AmdModule` export in the project. And I expect that to be very rare.